go/vt/vttablet/tabletserver/tabletserver.go

```diff
@@ -208,6 +208,43 @@ func NewServer(topoServer *topo.Server, alias topodatapb.TabletAlias) *TabletSer
 	return NewTabletServer(tabletenv.Config, topoServer, alias)
 }
 
+type TxPoolController interface {
```

Comment:
I don't think we need this interface. We should introduce interfaces only if it's necessary, like if there is a need to replace teCtrl with something else. Otherwise, directly using TxEngine is generally more readable.

Reply:
FWIW I disagree. In my opinion interfaces more clearly delineates how different parts of the code interact and make it more obvious when you add a new dependency across a code base. It is one of the most powerful tools we humans have to tackle complexity by breaking down code in smaller parts so that we can understand each part separately without having to understand everything at the same time.

Sugu, you might be able to keep all of Vitess in your head at the same time. Us mortals are not as fortunate. We have to page different parts in and out.

:-)

Reply:
In this particular case, I won't insist if you think the interface helps.

The reason why I dislike interfaces is because of the opposite reason: I can't hold the codebase in my head :).
Here's a classic example: https://github.com/vitessio/vitess/blob/master/go/vt/vtgate/scatter_conn.go#L146. This particular line calls into queryservice. As a challenge, you can try drilling down to figure what code path gets eventually used to call vttablet.
Another similar situation is with TabletStats that report health. I still don't understand how that mechanism works, and this has prevented me from fixing issues in that area. I know the health check mechanism is broken; People keep reporting strange issues. But there's nothing I can do there. Those interface abstractions make it impossible for me to understand what calls what.

Short story long: in this case, it's fairly obvious one way or another.